packages/block-editor/src/components/media-upload/index.native.js

```diff
@@ -270,13 +257,26 @@ export class MediaUpload extends Component {
 		}
 
 		const getMediaOptions = () => (
-			<Picker
-				title={ pickerTitle }
-				hideCancelButton
-				ref={ ( instance ) => ( this.picker = instance ) }
-				options={ this.getMediaOptionsItems() }
-				onChange={ this.onPickerSelect }
-			/>
+			<View>
+				{ this.state.isAddFromUrlBottomsheetVisible && (
+					<AddFromUrlBottomSheet
+						onClose={ ( { url } ) => {
+							this.setState( {
+								isAddFromUrlBottomsheetVisible: false,
+							} );
```

Comment:
I'm not sure I like handling the state for the bottom sheet here. I wanted to encapsulate it within the `AddFromUrlBottomSheet` the issue I was having was the `AddFromUrlBottomSheet` is a function component while the `MediaUpload` is a class component, so the ref ideas I had weren't clear to me. Let me know what you think. This sheet is based on https://github.com/WordPress/gutenberg/pull/31343